hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/hudi/execution/RangeSample.scala

```diff
@@ -258,7 +258,7 @@ object RangeSampleSort {
     val fieldNum = df.schema.fields.length
     val checkCols = orderByCols.filter(col => columnsMap(col) != null)
 
-    if (orderByCols.isEmpty || checkCols.isEmpty) {
+    if (orderByCols.isEmpty || checkCols.isEmpty || df.take(2).length < 2L) {
```

Comment:
What abt to check rangeBounds' size? This way we don't need to read dataframe to get result again.

Reply:
yes makes sense. Also added a test case